lib/linguist/generated.rb

```diff
@@ -533,5 +534,12 @@ def generated_grpc_cpp?
       return false unless lines.count > 1
       return lines[0].start_with?("// Generated by the gRPC")
     end
+
+    # Internal: Is this a relay-compiler generated graphql.js file?
+    #
+    # Return true or false
+    def generated_graphql_relay_js?
+      !!name.match(/__generated__\/[_A-Za-z][_0-9A-Za-z]+\.graphql\.js$/)
```

Comment:
Shouldn't we mark the whole `__generated__` directory as generated?

Reply:
Isn't the `__generated__` directory created by relay-compiler?

Reply:
I'd probably mark the whole `__generated__` directory as generated, but then I don't know much about GraphQL and it looks like you have some reservations about it. Anything specific that makes you think this could be "unsafe" (i.e., not the expected user behavior)?

Reply:
That sounds good to me. In a GraphQL relay project it is definitely safe to mark all files under the `__generated__` directory as generated. I'm not sure about other languages/frameworks, and if they ever would use a `__generated__` directory for some other purpose... yeah probably never...

One sec will make the change

Reply:
From what I could see in the search results, it looks like it's not really used outside of GraphQL. Worst-case scenario, we'll reverse if we have users complaining :/